arches/app/search/components/resource_type_filter.py

```diff
@@ -18,12 +18,19 @@
 
 
 def get_permitted_graphids(permitted_nodegroups):
-    return {
-        str(graph_id)
-        for graph_id in Node.objects.filter(nodegroup__in=permitted_nodegroups)
-        .values_list("graph_id", flat=True)
-        .distinct()
-    }
+    permitted_graphids = set()
+
+    for node in Node.objects.filter(nodegroup__in=permitted_nodegroups):
+        permitted_graphids.add(str(node.graph_id))
+
+    permitted_graphids = set(
+        str(graphid)
+        for graphid in GraphModel.objects.filter(graphid__in=permitted_graphids)
+        .exclude(publication=None)
+        .values_list("graphid", flat=True)
+    )
+
+    return permitted_graphids
```

Comment:
I think it would be clearer to use different variable names rather than to reassign a different value to the same variable name. Something like: 
```suggestion
    permitted_and_published_graphids = set(
        str(graphid)
        for graphid in GraphModel.objects.filter(graphid__in=permitted_graphids)
        .exclude(publication=None)
        .values_list("graphid", flat=True)
    )

    return permitted_and_published_graphids
```